File: talks/2016-devcon-original/libgeth/inproc.java
NodeConfig conf = Geth.newNodeConfig();
Node node = Geth.newNode(getFilesDir() + "/.ethereum", conf); // HL
node.start(); // HL

NodeInfo info = node.getNodeInfo(); // HL
textbox.append("My name: " + info.getName() + "\n");
textbox.append("My address: " + info.getListenerAddress() + "\n");
textbox.append("My protocols: " + info.getProtocols() + "\n\n");

EthereumClient ec = node.getEthereumClient(); // HL
textbox.append("Latest block: " + ec.getBlockByNumber(ctx, -1).getNumber());

File: talks/2016-devcon-original/libgeth/inproc.go
package main

import (
	"fmt"

	"github.com/ethereum/go-ethereum/cmd/utils"
	"github.com/ethereum/go-ethereum/eth"
	"github.com/ethereum/go-ethereum/ethclient"
	"github.com/ethereum/go-ethereum/node"
	"golang.org/x/net/context"
)

var ctx = context.Background()

// START OMIT
func main() {
	stackConf := &node.Config{DataDir: "/home/karalabe/.ethereum", ListenAddr: ":0"}
	stack, _ := node.New(stackConf) // HL

	ethConf := &eth.Config{ChainConfig: utils.MainnetChainConfig}
	_ = stack.Register(func(ctx *node.ServiceContext) (node.Service, error) { // HL
		return eth.New(ctx, ethConf)
	})

	_ = stack.Start() // HL
	fmt.Println("My datadir:", stack.DataDir())
	fmt.Println("My address:", stack.Server().NodeInfo().ListenAddr, "\n")

	api, _ := stack.Attach() // HL
	block, _ := ethclient.NewClient(api).BlockByNumber(ctx, nil)
	fmt.Println("Latest block:", block.Number())
}

// END OMIT
